coxeter/shapes/polyhedron.py

```diff
@@ -356,9 +357,45 @@ def vertices(self):
 
     @property
     def faces(self):
-        """list(:class:`numpy.ndarray`): Get the polyhedron's faces."""
+        """list(:class:`numpy.ndarray`): Get the polyhedron's faces.
+
+        Results returned as vertex index lists.
+        """
         return self._faces
 
+    @cached_property
+    def edges(self):
+        """set(tuple(int,int)): Get the polyhedron's edges.
+
+        Results returned as vertex index pairs,  with each edge of the polyhedron
+        included exactly once.  Edge (i,j) pairs are ordered by vertex index with i<j.
+        """
+        ij_pairs = np.array(
+            [
+                [i, j]
+                for face in self.faces
+                for i, j in zip(face, np.roll(face, -1))
+                if i < j
+            ]
+        )
+        sorted_indices = np.lexsort(ij_pairs.T[::-1])
+        sorted_ij_pairs = ij_pairs[sorted_indices]
+        # Make edge data read-only so that the cached property of this instance
+        # cannot be edited
+        sorted_ij_pairs.flags.writeable = False
+
+        return sorted_ij_pairs
+
+    @property
+    def edge_vectors(self):
+        """list(tuple(float,float,float)): Get the polyhedron's edges as vectors."""
+        return self.vertices[self.edges[:, 1]] - self.vertices[self.edges[:, 0]]
+
+    @property
+    def num_edges(self):
+        """int: Get the number of edges."""
+        return len(self.edges)
```

Comment:
This can be calculated without the need to compute `self.edges` at all. `F + V = E + 2`, per Euler's formula. 😄 I'd suggest:
```suggestion
        return self.num_vertices + self.num_faces - 2
```
This avoids doing the face iterations for complex shapes.

Reply:
This is only strictly true for convex shapes - I have added a method to convex_polyhedron that makes use of the Euler characteristic, and have added additional pytests (see [d3f5203](https://github.com/glotzerlab/coxeter/pull/171/commits/d3f520379977f239033fb7351039d9336884b3ef))

Reply:
Good point. Thanks for catching that. I’ll resolve this.